enhancements/api-review/add-ImagePolicy-to-config.openshift.io-v1alpha1.md

```diff
@@ -0,0 +1,294 @@
+	Images []ImageScope `json:"images"`
+	// policy defines the verification policy
+	Policy Policy `json:"policy"`
+}
+
+// ImageScope is the item of the images list.
+// https://go.dev/play/p/FfL_3bpqSX5
+// +kubebuilder:validation:Pattern=`^\*(?:\.(?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9]))+$|^((?:(?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9])(?:(?:\.(?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9]))+)?(?::[0-9]+)?/)?[a-z0-9]+(?:(?:(?:[._]|__|[-]*)[a-z0-9]+)+)?(?:(?:/[a-z0-9]+(?:(?:(?:[._]|__|[-]*)[a-z0-9]+)+)?)+)?)(?::([\w][\w.-]{0,127}))?(?:@([A-Za-z][A-Za-z0-9]*(?:[-_+.][A-Za-z][A-Za-z0-9]*)*[:][[:xdigit:]]{32,}))?$`
+type ImageScope string
+
+type Policy struct {
+	// oidcIssuer contains the expected OIDC issuer.
+	// Example: "https://expected.OIDC.issuer/"
+	// Required if fulcioCAData is set.
+	// subjectEmail must be set and keyData must be empty if oidcIssuer is set
+	// +optional
+	OIDCIssuer string `json:"oidcIssuer,omitempty"`
+
+	// subjectEmail holds the email address of the subject.
+	// Example: "expected-signing-user@example.com"
+	// Required if fulcioCAData is set.
+	// oidcIssuer must be set and keyData must be empty if subjectEmail is set.
+	// +optional
+	SubjectEmail string `json:"subjectEmail,omitempty"`
+
+	// keyData contains inline base64 data of the public key.
+	// keyData can be empty if the image got signed keyless.
+	// Requires oidcIssuer, subjectEmail, and fulcioCAData to be empty if keyData is set.
+	// +optional
+	KeyData string `json:"keyData,omitempty"`
+
+	// fulcioCAData contains inline base64 data for the fulcio CA certificate.
+	// +optional
+	FulcioCAData string `json:"fulcioCAData,omitempty"`
+
+	// rekorKeyData contains inline base64 data of the rekor public key.
+	// Required if fulcioCAData is set.
+	// +optional
+	RekorKeyData string `json:"rekorKeyData,omitempty"`
+
+	// signedIdentity specifies what image identity the signature.
+	// claims about the image.
+	// +optional
+	SignedIdentity Identity `json:"signedIdentity,omitempty"`
+}
+
+type Identity struct {
+	// Defaults to "MatchRepoDigestOrExact"
+	IdentityMatchPolicy IdentityMatchPolicy `json:"identityMatchPolicy,…"`
+	DockerRepository    string              `json:"dockerRepository,omitempty"`
+	Prefix string `json:"prefix,omitempty"`
+	SignedPrefix string `json:"signedPrefix,omitempty"`
+}
+
+type IdentityMatchPolicy string
+
+const (
+	IdentityMatchPolicyMatchRepoDigestOrExact = "MatchRepoDigestOrExact"
+	IdentityMatchPolicyMatchRepository        = "MatchRepository"
+	IdentityMatchPolicyExactRepository        = "ExactRepository"
+	IdentityMatchPolicyRemapIdentity          = "RemapIdentity"
+)
+
+type ImagePolicyStatus struct {
+	// PolicyJSON contains the whole policy applied to the namespace
+	// which got written to disk. This includes cluster-wide policies
+	// from the `openshift-config` namespace as well.
+	PolicyJSON string `json:"policyJSON,omitempty"`
```

Comment:
The main impact of an ImagePolicy is a `/path/to/policies/<NAMESPACE>.json` with this JSON content.  But `use-sigstore-attachments` entries will also be written to `/etc/containers/registries.d/*.yaml`.  Should that information be represented in `status`?  Or should folks interested in that content be checking the rendered MachineConfig `spec` to see how it's working?

Reply:
`/etc/containers/registries.d/*.yaml` turns on the read of sigstore signatures. It seems the content will not be frequently used like the global policy `policyJson` unless for debugging.
We don't  have to have separate yaml file for each  namespace. Folks can check the mahcineconfig or check the single yaml file like `/etc/containers/registries.d/sigstore.yaml`